PARPACK/SRC/MPI/pcgetv0.f

```diff
@@ -213,26 +219,30 @@ subroutine pcgetv0
 c     | random number generator           |
 c     %-----------------------------------%
 c
+      if (inits) then
 c
-c     %-----------------------------------%
-c     | Generate a seed on each processor |
-c     | using process id (myid).          |
-c     | Note: the seed must be between 1  |
-c     | and 4095.  iseed(4) must be odd.  |
-c     %-----------------------------------%
+c        %-----------------------------------%
+c        | Generate a seed on each processor |
+c        | using process id (myid).          |
+c        | Note: the seed must be between 1  |
+c        | and 4095.  iseed(4) must be odd.  |
+c        %-----------------------------------%
 c
-      call MPI_COMM_RANK(comm, myid, ierr)
-      igen = 1000 + 2*myid + 1
-      if (igen .gt. 4095) then
-         write(0,*) 'Error in p_getv0: seed exceeds 4095!'
-      end if
+         call MPI_COMM_RANK(comm, myid, ierr)
+         igen = 1000 + 2*myid + 1
+         if (igen .gt. 4095) then
+            write(0,*) 'Error in p_getv0: seed exceeds 4095!'
+         end if
+c
+         iseed(1) = igen/1000
+         igen     = mod(igen,1000)
+         iseed(2) = igen/100
+         igen     = mod(igen,100)
+         iseed(3) = igen/10
+         iseed(4) = mod(igen,10)
 c
-      iseed(1) = igen/1000
-      igen     = mod(igen,1000)
-      iseed(2) = igen/100
-      igen     = mod(igen,100)
-      iseed(3) = igen/10
-      iseed(4) = 7
+         inits = .false.
```

Comment:
AFAIU, once `inits` is back to `.false.`, nothing makes it back to `.true.`!?...
"Who" can or need to make `inits` back to `.true.`, when and why?!... Not clear to me: could you detail this https://github.com/opencollab/arpack-ng/pull/414#issuecomment-1501563595 (and possibly push comments about this in the code to keep track of the logic)? Once this is clear to me, I may have a look at the MPI part and find a way to make CI robust.

Reply:
The purpose of `inits` is to ensure that the RNG state is set only once, and after that it is propagated correct. `inits` will be `.true.` on startup. Then as soon as the RNG is initialized/seeded, it is set to `.false.` to prevent future re-initialization. It will never be set to `.true.` again during the program's life time.

This logic is from the original ARPACK.

Can we please merge the fix for the serial part before handling the parallel one? I can separate the parallel one into a different PR. Merging the serial fix is a priority for us.

Reply:
I'd like to find a way to fix this for both serial/parallel otherwise parallel will never be fixed (not done today = never done...). I'll have a look when possible.